storage/ctab/src/main/java/org/openscience/cdk/io/MDLV3000Writer.java

```diff
@@ -445,15 +448,54 @@ private void writeBondBlock(IAtomContainer mol,
                 endIdx = tmp;
             }
 
-            final int order = bond.getOrder() == null ? 0 : bond.getOrder().numeric();
-
-            // TODO add query bonds
-            if (order < 1 || order > 3)
+            int order = bond.getOrder() == null ? 0 : bond.getOrder().numeric();
+            int bondType = -1; // initialize to satisfy compiler, this value is not expected to actually being written
+            // If bond is an object of type QueryBond its bond order is set to null, so the variable 'order' ends up being 0.
+            // Aromatic bonds have a (1) bond order of IBond.Order.UNSET (which also yields a value of 0 for 'order') and
+            // (2) the flag IChemObject.AROMATIC set to true.
+            if (order == 0) {
+                if (bond.getOrder() == IBond.Order.UNSET && bond.getFlag(IChemObject.AROMATIC)) {
+                    bondType = 4;
+                } else if (bond instanceof IQueryBond) {
+                    // Only query bonds of the class QueryBond are supported as the actual query expression needs to be
+                    // extracted from the bond.
+                    if (!(bond instanceof QueryBond)) {
+                        throw new CDKException("Query bond of type " + bond.getClass() + " cannot be written to V3000");
+                    }
+                    final Expr expression = ((QueryBond) bond).getExpression();
+                    // Expression are predicate trees and can be arbitrarily complex.
+                    // Only simple expressions consisting of a single node with a select few types indicating
+                    // a bond order are considered here.
+                    if (expression.left() != null || expression.right() != null) {
+                        throw new CDKException("Query bonds whose features are described by more than a single Expressions " +
+                                bond.getOrder() + " cannot be written to V3000");
+                    }
+                    switch (expression.type()) {
+                        case SINGLE_OR_DOUBLE:
```

Comment:
1) Should also check the OR(SINGLE,DOUBLE) OR(DOUBLE,SINGLE) , OR(SINGLE,AROM) etc. variants as they can technically occur if a user constructs them. 

Technically something daft like OR(DOUBLE,OR(SINGLE,FALSE)) is also equivalent. It would be easier if to a "canonical" expression but we don't have a way to do that. One option is you generate bond code which tells you what you match (let me have a go at writing a small function which does that). 

I also think bondType=8 should be the default in non-strict mode, this is common in other tools. "There is a query

Reply:
> I also think bondType=8 should be the default in non-strict mode, this is common in other tools.

I remember seeing `IChemObjectReader.Mode.{Relaxed|Strict}` but I cannot find anything analogous in `IChemObjectWriter`. Not sure, if I understand what you referring to here.